data_model/src/lib.rs

```diff
@@ -1002,6 +1011,22 @@ pub mod model {
         /// in the next request to continue fetching results of the original query
         pub cursor: crate::query::cursor::ForwardCursor,
     }
+
+    impl ChainId {
+        /// Create new [`Self`]
+        pub fn new(inner: &str) -> Self {
+            Self(inner.into())
+        }
+    }
+}
+
+impl Decode for ChainId {
+    fn decode<I: parity_scale_codec::Input>(
+        input: &mut I,
+    ) -> Result<Self, parity_scale_codec::Error> {
+        let boxed: String = parity_scale_codec::Decode::decode(input)?;
+        Ok(Self::new(&boxed))
+    }
 }
```

Comment:
This could be improved by using constructor `fn new(inner: impl Into<Box<str>>)`.
I will extract it into separate good first issue.